bugwarrior/services/trello.py

```diff
@@ -0,0 +1,159 @@
+        SHORTURL: {'type': 'string', 'label': 'Trello short URL'},
+        URL: {'type': 'string', 'label': 'Trello URL'},
+    }
+    UNIQUE_KEY = (CARDID,)
+
+    def get_default_description(self):
+        """ Return the old-style verbose description from bugwarrior.
+        """
+        return self.build_default_description(
+            title=self.record['name'],
+            url=self.record['shortUrl'],
+            number=self.record['idShort'],
+            cls='task',
+        )
+
+    def get_tags(self, twdict):
+        tmpl = Template(
+            self.origin.get('label_template', DEFAULT_LABEL_TEMPLATE))
+        return [tmpl.render(twdict, label=label['name'])
+                for label in self.record['labels']]
+
+    def to_taskwarrior(self):
+        twdict = {
+            'project': self.extra['boardname'],
+            'priority': 'M',
+            self.NAME: self.record['name'],
+            self.CARDID: self.record['id'],
+            self.BOARD: self.extra['boardname'],
+            self.LIST: self.extra['listname'],
+            self.SHORTLINK: self.record['shortLink'],
+            self.SHORTURL: self.record['shortUrl'],
+            self.URL: self.record['url'],
+        }
+        if self.origin['import_labels_as_tags']:
+            twdict['tags'] = self.get_tags(twdict)
+        return twdict
+
+
+class TrelloService(IssueService):
+    ISSUE_CLASS = TrelloIssue
+    # What prefix should we use for this service's configuration values
+    CONFIG_PREFIX = 'trello'
+
+    @classmethod
+    def validate_config(cls, config, target):
+        def check_key(opt):
+            """ Check that the given key exist in the configuration  """
+            key = cls._get_key(opt)
+            if not config.has_option(target, key):
+                die("[{}] has no '{}'".format(target, key))
+        super(TrelloService, cls).validate_config(config, target)
+        check_key('token')
+        check_key('api_key')
+        check_key('board')
+
+    def get_service_metadata(self):
+        """
+        Return extra config options to be passed to the TrelloIssue class
+        """
+        return {
+            'import_labels_as_tags':
+            self.config_get_default('import_labels_as_tags', False, asbool),
+            'label_template':
+            self.config_get_default('label_template', DEFAULT_LABEL_TEMPLATE),
+            }
+
+    def issues(self):
+        """
+        Returns a list of dicts representing issues from a remote service.
+        """
+        # First, we get the board name
+        board_id = self.config_get('board')
+        board = self.api_request(
+            "/1/boards/{board_id}".format(board_id=board_id), fields='name')
+        for lst in self.get_lists(board_id):
+            listextra = dict(boardname=board['name'], listname=lst['name'])
+            for card in self.get_cards(lst['id']):
+                yield self.get_issue_for_record(card, extra=listextra)
```

Comment:
Would you mind adding a test for this method? These lines appear to be the only ones not covered so far.